astropy/coordinates/stokes_coord.py

```diff
@@ -0,0 +1,238 @@
+
+        Returns
+        -------
+        arrays : list of ndarray
+        """
+        return [self._parent._data]
+
+
+class StokesCoord(ShapedLikeNDArray):
+    """
+    A representation of stokes coordinates with helpers for converting to profile names.
+
+    Parameters
+    ----------
+    stokes_values : array-like
+        The numeric values representing stokes coordinates.
+    """
+
+    info = StokesCoordInfo()
+
+    def __init__(self, stokes, copy=False):
+        stokes = np.asanyarray(stokes)
+        if stokes.dtype.kind == "U":
+            self._data = self._from_symbols(stokes)
+        else:
+            self._data = stokes
+
+    @property
+    def shape(self):
+        return self._data.shape
+
+    @property
+    def value(self):
+        return self._data
+
+    # def __array__(self):
+    #     return self._data
+
+    def _apply(self, method, *args, **kwargs):
+        cls = type(self)
+
+        if callable(method):
+            new = cls(method(self._data, *args, **kwargs))
+        else:
+            new = cls(getattr(self._data, method)(*args, **kwargs))
+
+        # Copy other 'info' attr only if it has actually been defined.
+        # See PR #3898 for further explanation and justification, along
+        # with Quantity.__array_finalize__
+        if "info" in self.__dict__:
+            new.info = self.info
+
+        return new
+
+    @staticmethod
+    def _from_symbols(symbols):
+        """
+        Construct a StokesCoord from strings representing the stokes symbols.
+        """
+        values_array = np.full_like(symbols, np.nan, dtype=float, subok=False)
+        for stokes_value, symbol in STOKES_VALUE_SYMBOL_MAP.items():
+            values_array[symbols == symbol.symbol] = stokes_value
+
+        if (nan_values := np.isnan(values_array)).any():
+            raise ValueError(
+                f"Unknown stokes symbols present in the input array: {np.unique(symbols[nan_values])}"
+            )
+
+        return values_array
+
+    @property
+    def _stokes_values(self):
+        """
+        A representation of the coordinate as integers.
+        """
+        # Note we unbroadcast and re-broadcast here to prevent the new array
+        # using more memory than the old one.
+        return np.broadcast_to(np.round(unbroadcast(self._data)), self.shape)
+
+    @property
+    def symbol(self):
+        """
+        The coordinate represented as strings
+        """
+        known_symbols = tuple(
+            ["?"] + [s.symbol for s in STOKES_VALUE_SYMBOL_MAP.values()]
+        )
+        max_len = np.max([len(s) for s in known_symbols])
+
+        # Note we unbroadcast and re-broadcast here to prevent the new array
+        # using more memory than the old one.
+        symbolarr = np.full(unbroadcast(self._data).shape, "?", dtype=f"<U{max_len}")
+
+        for value, symbol in STOKES_VALUE_SYMBOL_MAP.items():
+            symbolarr[unbroadcast(self._stokes_values) == value] = symbol.symbol
```

Comment:
This is the only place it is used - I'd suggest to just do the `np.round(unbroadcast(self._data))` in here